### DISCUSSION TITLE
WebGL state Diagram

### DISCUSSION THREAD

**Discussion by letochagone** [gfxfundamentals/webgl-fundamentals#421] [Q&A] [answered]
I added examples of the state diagram.
If I add examples it is so that I have a clear understanding of the fundamentals of webgl.
Here is my little problem. I added two examples that display the contents of a texture. The code is the same, except for loading the image: there are 2 syntaxes of Teximage2D.

first example:

![ok](https://user-images.githubusercontent.com/7129968/220837138-c67b9de3-e1fb-4106-9191-1ac0d1e3ea95.jpg)

well, it was a bit complicated to integrate the base64 character string. But the result is good. some pixels are a bit transparent, and it looks good

second release:

![wrong](https://user-images.githubusercontent.com/7129968/220837629-5484ce1c-0f8d-4d10-a9f8-b229fd3853d0.jpg)

**then it's strange, the texture display is different (the final display is ok).
In short, I wonder then if it is a bug, or if it is me who did not understand something**

**Comment by greggman** [comment, marked as answer]
The state diagram is not a generic debugger, it is not designed to handle any sample. It is designed only the handle the samples that are in it. If you add your own samples you'll have to modify the state diagram code to handle them.

**Reply by letochagone** [reply]
I added other examples that worked well. I just added the source code of these examples :

![image](https://user-images.githubusercontent.com/7129968/220858566-f48bd3f9-c923-4b30-baf4-cebdb6ba5f20.png)


![image](https://user-images.githubusercontent.com/7129968/220856045-04b93c15-1620-425a-8f23-495f99e59659.png)

these examples have been interpreted. I didn't have to change anything else.
that's why I thought that if this last example gave a curious result it was due to my misunderstanding of WebGL.

Besides, the example that worked well (i.e. the result corresponded to what I think I understood) is the example where the image is loaded (without using the ..onload() )
Which made me believe that the interpretation was powerful.

![image](https://user-images.githubusercontent.com/7129968/220856954-ccb9644f-4a67-4a49-a0c2-0e9e18a086e1.png)

![image](https://user-images.githubusercontent.com/7129968/220858228-d5d3dcb1-3996-4313-86d2-42bbde6d376e.png)


![image](https://user-images.githubusercontent.com/7129968/220858067-3980b210-d6ec-4a26-8e80-bf76e08d4ed3.png)


I was not using adding examples for the purpose of debugging, but for the purpose of understanding
